nameserver/nameserver_test.go

```diff
@@ -266,3 +332,192 @@ func TestTombstoneDeletion(t *testing.T) {
 	nameserver.deleteTombstones()
 	require.Equal(t, Entries{}, nameserver.entries)
 }
+
+// TestRestore tests whether entries have been restored after the nameserver restart
+// and after manually calling RestoreEntries.
+func TestRestore(t *testing.T) {
```

Comment:
I'm in favour of having multiple test cases which test separate features (i.e. unit testing).

However, I see an advantage of extending `TestNameservers` - it does some randomized testing which could test execution paths which were not exploited by `TestRestore`.